core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1915,7 +1915,8 @@ public String getUrlChildPrefix() {
      * @since 1.66
      * @see <a href="https://wiki.jenkins-ci.org/display/JENKINS/Hyperlinks+in+HTML">Hyperlinks in HTML</a>
      */
-    public @Nullable String getRootUrl() {
+    @CheckForNull
+    public String getRootUrl() {
```

Comment:
To answer the question in cffe9df0176b0ff895554ce7f2ea4d2f20062351: because after being set, it should not revert to null again, so some code could legitimately assume it is already non-null. But it is probably better to just force all code to handle nullness.